pkg/components/asap.go

```diff
@@ -115,9 +115,16 @@ func ASAPToken(_ context.Context, _ string, _ string, _ string) (interface{}, er
 	return &ASAPTokenComponent{}, nil
 }
 
+// NewComponent populates an ASAPToken with defaults.
+func NewComponent() *ASAPTokenComponent {
+	return &ASAPTokenComponent{}
+}
+
 // Settings generates a config populated with defaults.
 func (m *ASAPTokenComponent) Settings() *ASAPTokenConfig {
-	return &ASAPTokenConfig{}
+	return &ASAPTokenConfig{
+		TTL: time.Minute * 60,
```

Comment:
As a general principle, I prefer the identity provider (in this case ASAP) to dictate expiration for auth tokens rather than granting any arbitrary expiration duration as requested by client. This is, however, ultimately for ASAP team to decide.
In our case, for this PR, I prefer us not to set default expiration to 1 hour and instead force developer to set appropriate duration for their use case.
Another option would be that if micros provides some sort env variable for token expiration, we could use that as default.

Reply:
Yeah I agree with this.  You've convinced me; I think the lib should not set a default, but should keep the validation, which is by itself sufficient for the goal I was trying to achieve in this PR.  I'll remove the default.